common/ferrostar/src/routing_adapters/osrm/mod.rs

```diff
@@ -35,21 +37,35 @@ impl OsrmResponseParser {
 }
 
 impl RouteResponseParser for OsrmResponseParser {
-    fn parse_response(&self, response: Vec<u8>) -> Result<Vec<Route>, RoutingResponseParseError> {
+    fn parse_response(&self, response: Vec<u8>) -> Result<Vec<Route>, OsrmParsingError> {
         let res: RouteResponse = serde_json::from_slice(&response)?;
-        let via_waypoint_indices: HashSet<_> = res
-            .routes
+
+        // This isn't the most functional in style, but it's a bit difficult to construct a pipeline
+        // today. Stabilization of try_collect may help.
```

Comment:
I think there's a way to do this but I can look into that on my own time.

The thing I failed to notice though is that we just silently discard all errors 😂 I think that's on me, and I think it's probably wrong; we should bail on errors.

Reply:
So TIL you don't actually need `try_collect` to do what you want here! And, serendipitously, I actually needed this exact same pattern on some code I was working on over the weekend. The solution is described in [this comment](https://github.com/rust-lang/rust/issues/94047#issuecomment-1041432644) rather tersely with some code. What isn't obvious is that you can actually write your return type as `Result<Vec<_>, _>` and `collect` will do the right thing.